oak_restricted_kernel/src/syscall/dice_data.rs

```diff
@@ -22,13 +22,24 @@ use oak_restricted_kernel_interface::{Errno, DICE_DATA_FD};
 
 struct DiceDataDescriptor {
     data: DiceData,
+    index: usize,
+}
+
+impl DiceDataDescriptor {
+    fn new(data: DiceData) -> Self {
+        Self { index: 0, data }
+    }
 }
 
 impl FileDescriptor for DiceDataDescriptor {
     fn read(&mut self, buf: &mut [u8]) -> Result<isize, oak_restricted_kernel_interface::Errno> {
-        let data_as_bytes = <DiceData as zerocopy::AsBytes>::as_bytes(&self.data);
-        let length = min(data_as_bytes.len(), buf.len());
-        buf.copy_from_slice(&data_as_bytes[..length]);
+        let data_as_bytes = <DiceData as zerocopy::AsBytes>::as_bytes_mut(&mut self.data);
+        let length = min(data_as_bytes.len() - self.index, buf.len());
+        let slice_to_read = &mut data_as_bytes[self.index..length];
```

Comment:
Looking at it again, I don't think this works. I think it should be `[self.index..(self.index+length)]`, or the previous line could be changed to `let end = min(data_as_bytes.lent(), self.index + buf.len));`, and then this just becomes `[self.index..end]`